### DISCUSSION TITLE
In led_strip; purpose of  rmt_translator_set_context()?

### DISCUSSION THREAD

**Discussion by Xylopyrographer** [UncleRus/esp-idf-lib#520] [Q&A] [answered]
In the [led_strip driver](https://github.com/UncleRus/esp-idf-lib/blob/master/components/led_strip/led_strip.c#LL182C11-L182C25), [`led_strip_init()`](https://github.com/UncleRus/esp-idf-lib/blob/b867bd5cdc464abdf41c5912c481496ac40f0a39/components/led_strip/led_strip.c#LL182C11-L182C25) winds up calling  [`rmt_translator_set_context()`](https://github.com/UncleRus/esp-idf-lib/blob/b867bd5cdc464abdf41c5912c481496ac40f0a39/components/led_strip/led_strip.c#LL202C62-L202C62) (if the IDF version falls within the range where the `brightness` of the strip is enabled).

I can't figure out what and am hoping someone can explain what exactly `rmt_translator_set_context()` does and why the `strip` descriptor is passed in the [`CHECK(rmt_translator_set_context(config.channel, strip));`](https://github.com/UncleRus/esp-idf-lib/blob/b867bd5cdc464abdf41c5912c481496ac40f0a39/components/led_strip/led_strip.c#L202) call?

**Comment by UncleRus** [comment, marked as answer]
The purpose of the rmt_translator_set_context() call is obvious: this function sets the strip handle as the translator context. Later, in the _rmt_adapter() function, which is called by the translator, we get this descriptor using rmt_translator_get_context(), find out the contents of the "brightness" field and automatically apply it using scale8_video().

**Comment by Xylopyrographer** [comment]
Hello and thank-you for the reply. Appreciate you taking the time.

Have a great one.

On May 29, 2023, at 8:33 AM, Ruslan V. Uss ***@***.***> wrote:


The purpose of the rmt_translator_set_context() call is obvious: this function sets the strip handle as the translator context. Later, in the _rmt_adapter() function, which is called by the translator, we get this descriptor using rmt_translator_get_context(), find out the contents of the "brightness" field and automatically apply it using scale8_video().
